src/Meshes/interval.jl

```diff
@@ -167,34 +191,48 @@ function IntervalMesh(
     domain::IntervalDomain{CT},
     stretch::GeneralizedExponentialStretching{FT};
     nelems::Int,
+    reverse_mode::Bool = false,
 ) where {CT <: Geometry.Abstract1DPoint{FT}} where {FT}
     if nelems ≤ 1
         throw(ArgumentError("`nelems` must be ≥ 2"))
     end
+
     dz_bottom, dz_top = stretch.dz_bottom, stretch.dz_top
-    if !(dz_bottom ≤ dz_top)
+    if !(dz_bottom ≤ dz_top) && !reverse_mode
```

Comment:
Alternatively, we could leave this as

```julia
    if !(dz_bottom ≤ dz_top)
        throw(ArgumentError("dz_bottom must be ≤ dz_top"))
    end
```
and move it to below where `dz_bottom` and `dz_top` are recomputed, that may actually be simpler, what do you think?